### PR TITLE
load item product fix

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
There was redundant complexity in loading item product that led to this error. Please check the Asana task for the context

Fixes:
https://app.asana.com/0/474713171217346/1141424540626593/f

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [x] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [ ] I have created or modified unit tests to sufficiently cover my changes.

### CODE REVIEW

File: Helper/Cart.php [hunk 1]
```diff
@@ -1028,7 +1028,7 @@ function ($item) use ($imageBlock, &$totalAmount, &$diff, $storeId) {
                 ///////////////////////////////////////////
                 // Get item attributes / product properties
                 ///////////////////////////////////////////
-                $item_options = $item->getProduct()->getTypeInstance(true)->getOrderOptions($item->getProduct());
+                $item_options =$_product->getTypeInstance(true)->getOrderOptions($_product);
```
Comment (roopakv) [inline Helper/Cart.php]: seems like `getTypeInstance` doesnt accept a param? or am i looking at the wrong m2

https://github.com/magento/magento2/blob/2b7761832406cb0875d337ef33401db36741c2c6/app/code/Magento/Catalog/Model/Product.php#L691

Comment (danac-gs) [reply Helper/Cart.php]: Correct. Fixed.

File: Helper/Cart.php [hunk 2]
```diff
@@ -1028,7 +1028,7 @@ function ($item) use ($imageBlock, &$totalAmount, &$diff, $storeId) {
                 ///////////////////////////////////////////
                 // Get item attributes / product properties
                 ///////////////////////////////////////////
-                $item_options = $item->getProduct()->getTypeInstance(true)->getOrderOptions($item->getProduct());
+                $item_options =$_product->getTypeInstance()->getOrderOptions($_product);
```
Comment (YevhenShBolt) [inline Helper/Cart.php]: need one space after `=`

Comment (danac-gs) [reply Helper/Cart.php]: Fixed. @YevhenShBolt please re-approve so I could merge.

**Pull request merged**